pallets/asset-registry/src/lib.rs

```diff
@@ -529,16 +663,33 @@ impl<T: Config> Pallet<T> {
 	/// Create asset for given name or return existing AssetId if such asset already exists.
 	pub fn get_or_create_asset(
 		name: Vec<u8>,
-		asset_type: AssetType<T::AssetId>,
-		existential_deposit: T::Balance,
+		asset_type: AssetType,
+		existential_deposit: Balance,
 		asset_id: Option<T::AssetId>,
+		is_sufficient: bool,
 	) -> Result<T::AssetId, DispatchError> {
-		let bounded_name: BoundedVec<u8, T::StringLimit> = Self::to_bounded_name(name)?;
+		let bounded_name = Self::try_into_bounded(Some(name))?;
```

Comment:
if you use the type boundedvec instead of Vec<u8>, you dont have problems and no need for the inconsistentstate error at all.